Examples/destination_plugins/AdjustDestination.swift

```diff
@@ -0,0 +1,210 @@
+//
+//  AdjustDestination.swift
+//  DestinationsExample
+//
+//  Created by Brandon Sneed on 5/27/21.
+//
+
+// NOTE: You can see this plugin in use in the DestinationsExample application.
+//
+// This plugin is NOT SUPPORTED by Segment.  It is here merely as an example,
+// and for your convenience should you find it useful.  Please contact Adjust
+// about providing full support and publishing if desired.
+//
+// Adjust SPM package can be found here: https://github.com/adjust/ios_sdk
+
+// MIT License
+//
+// Copyright (c) 2021 Segment
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+import Segment
+import Adjust
+
+internal struct AdjustSettings: Codable {
+    let appToken: String
+    let setEnvironmentProduction: Bool?
+    let setEventBufferingEnabled: Bool?
+    let trackAttributionData: Bool?
+    let setDelay: Bool?
+    let customEvents: JSON?
+    let delayTime: Double?
+}
+
+@objc
+public class AdjustDestination: NSObject, DestinationPlugin {
+    public let timeline: Timeline = Timeline()
+    public let type: PluginType = .destination
+    public let name: String
+    public var analytics: Analytics? = nil
+    
+    internal var settings: AdjustSettings? = nil
+    
+    @Atomic var started = false
+    
+    required public init(name: String) {
+        self.name = name
+    }
+    
+    public func update(settings: Settings) {
+        guard let settings: AdjustSettings = settings.integrationSettings(name: "Adjust") else { return }
+        self.settings = settings
+        
+        var environment = ADJEnvironmentSandbox
+        if let _ = settings.setEnvironmentProduction {
+            environment = ADJEnvironmentProduction
+        }
+        
+        let adjustConfig = ADJConfig(appToken: settings.appToken, environment: environment)
```

Comment:
Do we need to protect against an empty appToken? Can it be nil?

Reply:
It potentially could be, but let adjust report that error.